drivers/sensor/stmemsc/stmemsc.h

```diff
@@ -76,5 +107,36 @@ int stmemsc_spi_read(const struct spi_dt_spec *stmemsc,
 		     uint8_t reg_addr, uint8_t *value, uint8_t len);
 int stmemsc_spi_write(const struct spi_dt_spec *stmemsc,
 		      uint8_t reg_addr, uint8_t *value, uint8_t len);
+
+/*
+ * Populate the stmdev_ctx_t structure pointed by stmdev_ctx_ptr with
+ * specific stmemsc sp APIs that set reg_addr bit6 to '1' in order to allow
+ * multiple read/write operations. This is common in some STMEMSC drivers
+ */
+#define STMEMSC_CTX_SPI_INCR(stmdev_ctx_ptr)				\
+	.ctx = {							\
+		.read_reg = (stmdev_read_ptr) stmemsc_spi_read_incr,	\
+		.write_reg = (stmdev_write_ptr) stmemsc_spi_write_incr,	\
+		.mdelay = (stmdev_mdelay_ptr) stmemsc_mdelay,		\
```

Comment:
odd... does this work?! `stmemsc_mdelay` is defined as `inline` so you should not be able to dereference it

Reply:
That's a good point! I think that it works just because the compiler is smart enough to assign in this way:

```.mdelay = (stmdev_mdelay_ptr) k_msleep,		\```

If I'm not wrong, the 'inline' clause is just an hint to the compiler for optimization. If a function address is required, the compiler will generate a non-inline version of the function. Does it make sense to you?

Reply:
yeah I think it makes sense, guess `always_inline` is what would make it fall apart, that said, does it make sense to keep declaring it as inline? seems like it's always using with a reference